### PR TITLE
adds circleci config to run tests and deploy gem

### PR DESCRIPTION
**Pull request by charlie-healthfinch**
## Description of Change

Adds Circle CI config to

* run tests
* deploy gem

## Testing Procedures to Verify the Functionality from this Change

We know the tests work because you can see the output below. We aren't going to be able to test the gem deployment until we bump the version and merge to master.

### Deployment
- [s] Deployment follows standard procedure

If deployment does not follow standard procedure please explain the deployment process for this PR.

### Rollback
- [x] Rollback follows standard procedure

If rollback does not follow standard procedure please explain the deployment process for this PR.

## Process Checks

- [x] Story card has been accepted by requester

- [x] Local run of tests have been completed

- [ ] Has been deployed and tested in Stage

- [x] This was worked on in a pair

If any of the above boxes were not checked please explain why:

## Security

- [x] This Pull Request does not introduce a new break of any of our Informational Security Policies

- [x] This Pull Request does not create new ingress or egress rules, including opening any new ports

- [x] This Pull Request has been evaluated by all OWASP principles and does not introduce any gaps in security

If any of the above boxes were not checked please explain why:

### Other Comments

** If all checkbox answers are yes above this change is pre-approved by manager of the application and can be merged with a Pull Request approval. If any of the boxes are unchecked the manager responsible for the application has the ability to approve the change and take on the risk.

Co-authored-by: Paul Richardson <paul.richardson@healthfinch.com>

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  test:
+    docker:
+      - image: circleci/ruby:2.4.9-browsers
+        environment:
+          BUNDLE_JOBS: 2
+          BUNDLE_RETRY: 3
+          BUNDLE_PATH: vendor/bundle
+    steps:
+      - checkout
+      # Which version of bundler?
```
Comment (defndaines) [inline .circleci/config.yml]: We have been using the latest 1.x elsewhere, though if it works with 2.x and that doesn't cause a problem, go ahead.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      - checkout
+      - run:
+          name: Build Gem
+          command: gem build allscripts_unity_clientgemspec
+      - run:
+          name: Upload Gem to Rubygems
+          command: |
+            echo "gem `gem --version`"
+            mkdir ~/.gem
+            cat .circleci/gem_credentials | sed -e "s/__RUBYGEMS_API_KEY__/${RubygemsKey}/" > ~/.gem/credentials
```
Comment (defndaines) [inline .circleci/config.yml]:
```suggestion
            sed -e "s/__RUBYGEMS_API_KEY__/${RubygemsKey}/" .circleci/gem_credentials > ~/.gem/credentials
```

File: .circleci/config.yml (hunk 1 above)
Comment (charlie-healthfinch) [reply .circleci/config.yml]: It seemed to work okay. :)

**Comment by charlie-healthfinch** [comment]
@defndaines committed that change. :)

**Pull request merged**